### PR TITLE
Fix @param null on RequestInterface::getServer() $filter argument

### PR DESCRIPTION
**Pull request by samsonasik**
**Description**

Looking at the usage, the `$filter` can be null or integer, eg: 

```php
$referer = $_SESSION['_ci_previous_url'] ?? Services::request()->getServer('HTTP_REFERER', FILTER_SANITIZE_URL);
```

or

```php
$request->getServer(['SERVER_PROTOCOL', 'REQUEST_URI']);
```

so it seems can use `@param string|int|null` for it.

**Checklist:**
- [x] Securely signed commits

### CODE REVIEW

File: system/HTTP/RequestInterface.php [hunk 1]
```diff
@@ -40,8 +40,8 @@ public function isValidIP(string $ip, ?string $which = null): bool;
      * Fetch an item from the $_SERVER array.
      * Supplied by RequestTrait.
      *
-     * @param string $index  Index for item to be fetched from $_SERVER
-     * @param null   $filter A filter name to be applied
+     * @param string          $index  Index for item to be fetched from $_SERVER
+     * @param int|string|null $filter A filter name to be applied
```
Comment (kenjis) [inline system/HTTP/RequestInterface.php:44]: Why `string`?

**Comment by kenjis** [comment]
The `$filter` will be passed to `filter_var()`:
https://github.com/codeigniter4/CodeIgniter4/blob/2e1761337668b7c1c5887a619859fa9864ff4161/system/HTTP/RequestTrait.php#L268
and the parameter is `int`: https://www.php.net/filter_var

File: system/HTTP/RequestInterface.php (hunk 1 above)
Comment (samsonasik) [reply system/HTTP/RequestInterface.php:44]: I found the usage:

```
$request->getServer(['SERVER_PROTOCOL', 'REQUEST_URI']);
```

On find in code

Comment (kenjis) [reply system/HTTP/RequestInterface.php:44]: It it the first parameter as an array. The `$filter` is null.

File: system/HTTP/RequestInterface.php [hunk 2]
```diff
@@ -40,8 +40,8 @@ public function isValidIP(string $ip, ?string $which = null): bool;
      * Fetch an item from the $_SERVER array.
      * Supplied by RequestTrait.
      *
-     * @param string $index  Index for item to be fetched from $_SERVER
-     * @param null   $filter A filter name to be applied
+     * @param string          $index  Index for item to be fetched from $_SERVER
```
Comment (kenjis) [inline system/HTTP/RequestInterface.php:43]: `array|string|null`

File: system/HTTP/RequestInterface.php (hunk 1 above)
Comment (samsonasik) [reply system/HTTP/RequestInterface.php]: Ok, updated https://github.com/codeigniter4/CodeIgniter4/pull/7322/commits/d6c8e5a5eb96ae42c978ec5a9ab14b225c966742

File: system/HTTP/RequestInterface.php (hunk 2 above)
Comment (samsonasik) [reply system/HTTP/RequestInterface.php]: Ok, updated https://github.com/codeigniter4/CodeIgniter4/pull/7322/commits/d6c8e5a5eb96ae42c978ec5a9ab14b225c966742

**Review by kenjis — APPROVED** [review]
LGTM! Thank you.

**Review by paulbalandan — APPROVED** [review]
Need to rebase on top of latest develop to remove Kint changes.

**Comment by samsonasik** [comment]
@paulbalandan rebased :+1:

**Review by paulbalandan — APPROVED** [review]

**Comment by samsonasik** [comment]
@kenjis @paulbalandan thank you for the review

**Pull request merged**